modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/application/publisher/APIApplicationPublisherImpl.java

```diff
@@ -78,6 +83,8 @@ public void unpublish(String baseURL, long companyId) {
 				"APIApplicationPublisher not available");
 		}
 
+		_endpointMatchersMap.remove(_getOSGiJaxRsName(baseURL, companyId));
```

Comment:
Maybe better to remove the EndpointMatcher after unregistering the services to avoid possible weird race conditions (I know the returned code would be 404 but just in case).

Reply:
Done in the original commit.